### PR TITLE
ddl: use the correct schema version when waitSchemaSynced

### LINKED ISSUE #37209: tidb doesn't check the newest schema version if crushed before updating the global version
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

tidb doesn't check the newest schema version if crushed before updating the global version.
Then it may break the F1 online schema algorithm.

### 4. What is your TiDB version? (Required)
All versions.

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?

Issue Number: close #37209

Problem Summary:
tidb doesn't check the newest schema version if crushed before updating the global version.

### What is changed and how it works?

Use GetSchemaVersionWithNonEmptyDiff instead of mustGetGlobalVersion to get the newest schema version.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/db_test.go [hunk 1]
```diff
@@ -1714,3 +1714,15 @@ func TestBuildMaxLengthIndexWithNonRestrictedSqlMode(t *testing.T) {
...
+
+func TestTiDBDownBeforeUpdateGlobalVersion(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("create table t(a int)")
+
+	require.NoError(t, failpoint.Enable("github.com/pingcap/tidb/ddl/mockDownBeforeUpdateGlobalVersion", `return(true)`))
+	require.NoError(t, failpoint.Enable("github.com/pingcap/tidb/ddl/checkDownBeforeUpdateGlobalVersion", `return(true)`))
```
Comment (zimulala) [inline ddl/db_test.go:1726]: Please do `Disable` after using.

File: ddl/ddl_worker.go [hunk 2]
```diff
@@ -1299,19 +1312,34 @@ func (w *worker) waitSchemaChanged(ctx context.Context, d *ddlCtx, waitTime time
 // but in this case we don't wait enough 2 * lease time to let other servers update the schema.
 // So here we get the latest schema version to make sure all servers' schema version update to the latest schema version
 // in a cluster, or to wait for 2 * lease time.
-func (w *worker) waitSchemaSynced(d *ddlCtx, job *model.Job, waitTime time.Duration) {
+func (w *worker) waitSchemaSynced(d *ddlCtx, job *model.Job, waitTime time.Duration) error {
```
Comment (lcwangchao) [inline ddl/ddl_worker.go:1315]: Why add an error return? I found it always returns nil actually.

Comment (wjhuang2016) [reply ddl/ddl_worker.go:1315]: L1328 should return err

**Review by zimulala — APPROVED** [review]
LGTM

**Review by lcwangchao — APPROVED** [review]

**Pull request merged**